cmake/modules/FindASan.cmake

```diff
@@ -0,0 +1,13 @@
+set(ASan_LIB_NAME ASan)
+
+find_library(ASan_LIBRARY
+  NAMES libasan.so libasan.so.5 libasan.so.4 libasan.so.3 libasan.so.2 libasan.so.1 libasan.so.0
+  PATHS ${SANITIZER_PATH} /usr/lib64 /usr/lib /usr/local/lib64 /usr/local/lib ${CMAKE_PREFIX_PATH}/lib)
+
+include(FindPackageHandleStandardArgs)
+find_package_handle_standard_args(ASan DEFAULT_MSG
+  ASan_LIBRARY)
+
+mark_as_advanced(
+  ASan_LIBRARY
```

Comment:
Sorry, could you please explain why `ASan_LIBRARY` is here and in `REQUIRED_VARS` of `find_package_handle_standard_args()` at the same time?

Reply:
`find_package_handle_standard_args()` is the one that sets the variable `ASan_FOUND`. The `mark_as_advanced` is mostly cosmetic: it hides the variable `ASan_LIBRARY` from the CMake GUI, unless the advance mode is enabled. See https://stackoverflow.com/a/47866572

Reply:
OK, but **why** do we need to hide it?.. I think building with GTest will be done by quite experienced users so it's OK to have some additional variables to show in GUI.
Maybe just omit this part to simplify the code?

Reply:
Sure, let me try removing it and see if the sanitizer still works.